scipy/optimize/jonkervolgenant/jonkervolgenant.h

```diff
@@ -0,0 +1,47 @@
+/*
```

Comment:
@rgommers Is there a style guide for C++ around, and if so, how religiously is it followed? I've noticed that some of the existing code more or less follows the [NumPy C Style Guide](https://docs.scipy.org/doc/numpy/dev/style_guide.html) but is this something that's cared about? (If not, just ignore this comment.)

If so, some potential remarks:

* At least one line is longer than 80 chars.
* Function names and their opening braces don't start in the first column.
* Should the redistribution notice be there? It's not elsewhere.
* When part of a line break, `||` goes at the end of the first line rather than at the beginning of the second.
* Not all `if` branches have braces.
* One `for` lacks spaces after its `;`s.
* `shortestPathCosts` should maybe be `shortest_path_costs`.

On an unrelated note, I tried putting this through the VC++ profiler, and there appears to be no bottlenecks at all.

Reply:
There's no C++ style guide. For C we follow https://docs.scipy.org/doc/numpy/dev/style_guide.html, for C++ it's just about trying to stick to clean code and not using too much fancy C++ features just because you can.

Reply:
> On an unrelated note, I tried putting this through the VC++ profiler, and there appears to be no bottlenecks at all.

nice, thanks

Reply:
I have added braces in accordance with the NumPy C Style Guide.  The source files have also been formatted with: `clang-format -style="{BasedOnStyle: mozilla, IndentWidth: 4}"`
The variable names (and cases) reflect those in the pseudocode of the paper cited (DF Crouse).